src/ng/q.js

```diff
@@ -389,6 +399,10 @@ function qFactory(nextTick, exceptionHandler) {
       this.promise.$$state.value = reason;
       this.promise.$$state.status = 2;
       scheduleProcessQueue(this.promise.$$state);
+
+      if (this.promise.trackPromise) {
+        window.promises.angular.pendingCount--;
```

Comment:
Should this be evaluated at the same place as where we increment and decrement the queueSize variable? 
It seems that is a more accurate representation of this value.